spec/game_spec.rb

```diff
@@ -18,51 +18,149 @@ def print_message(string)
-      expect(console.print_message(console.printed_string)).to eq(expected_output)
+      expect(@console.print_message(@console.printed_string)).to eq(expected_output)
     end
 
     it 'marks a board appropriately with an X in the upper right corner' do
-      board = Board.new
-      console = FakeConsole.new
-      console.stub_get_input('3')
+      @console.stub_get_input(['3'])
 
-      game = Game.new(board, console)
-
-      game.play
+      @game.turn
 
       expected_output  = "   |   | X\n"
       expected_output += "-----------\n"
       expected_output += "   |   |  \n"
       expected_output += "-----------\n"
       expected_output += "   |   |  \n"
-      expect(console.print_message(console.printed_string)).to eq(expected_output)
+      expect(@console.print_message(@console.printed_string)).to eq(expected_output)
+    end
+  end
+
+  describe 'switch_players' do
+    it 'sets X as the current player when the game is started' do
+
+      expect(@game.switch_players).to eq('O')
+    end
+  end
+
+  describe 'play' do
+    it 'ends the game after X has a winning horizontal combination' do
+
+      @board.mark(0)
+      @board.mark(1)
+      @console.stub_get_input(['3'])
+      @game.play
+
+      expect(@rules.game_over?(@board)).to eq true
+    end
+  end
+
+  it 'ends the game after X has a winning diagonal combination' do
+
+    @board.mark(0)
+    @console.stub_get_input(['3', '4', '5', '7'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'ends the game after X has a winning diagonal combination on a full board' do
+
+    @console.stub_get_input(['3', '6', '9', '8', '7', '1', '4', '2', '5'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'ends the game after O has a winning horizontal combination' do
+
+    @console.stub_get_input(['1', '4', '7', '5', '8', '6'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'ends the game after O has a winning diagonal combination' do
+
+    @console.stub_get_input(['3', '1', '6', '5', '7', '9'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'ends the game after O has a winning diagonal combination on a full board' do
+
+    @console.stub_get_input(['2', '1', '4', '5', '3', '7', '6', '8', '9'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'ends the game after the board is full with no winning combinations' do
+
+    @console.stub_get_input(['1', '2', '3', '4', '5', '8', '6', '9', '7'])
+    @game.play
+
+    expect(@rules.game_over?(@board)).to eq true
+  end
+
+  it 'tells the user that the game is over when there is a winning combination' do
+
+    @console.stub_get_input(['3', '1', '6', '5', '7', '9'])
+    @game.play
+
+    expect(@console.print_message(@console.printed_string)).to eq('Game Over')
+  end
+
+  it 'tells the user that the game is over when there is a winning combination' do
```

Comment:
Duplicate tests names are hard to identify and debug. Make sure each of your tests has a unique name.